source/common/http/conn_manager_utility.cc

```diff
@@ -38,10 +39,11 @@ void ConnectionManagerUtility::mutateRequestHeaders(Http::HeaderMap& request_hea
   // peer. Cases where we don't "use remote address" include trusted double proxy where we expect
   // our peer to have already properly set XFF, etc.
   if (config.useRemoteAddress()) {
-    if (Network::Utility::isLoopbackAddress(connection.remoteAddress().c_str())) {
+    const std::string& remote_address = Network::Utility::hostFromUrl(connection.remoteAddress());
```

Comment:
hostFromUrl returns a temporary, so this should be std::string (not reference). Per above I realize this is terrible. I will fix this soon but for now let's just fix it up.